pkg/apm/components.go

```diff
@@ -0,0 +1,228 @@
+package apm
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// ListComponentsParams represents a set of filters to be
+// used when querying New Relic applications.
+type ListComponentsParams struct {
+	Name         string
+	IDs          []int
+	PluginID     int
+	HealthStatus bool
+}
+
+// ListComponents is used to retrieve the components associated with
+// a New Relic account.
+func (apm *APM) ListComponents(params *ListComponentsParams) ([]Component, error) {
```

Comment:
Not a blocker, but Jon and I had a [brief conversation](https://github.com/newrelic/newrelic-client-go/pull/55#discussion_r361521368) about returning an array of pointers. Any opposition to [this approach](https://github.com/newrelic/newrelic-client-go/blob/master/pkg/alerts/alert_conditions.go#L10-L11)? 

Some additional context as well:
> Question for you on `List` vs `Get`. List returns an array of Objects, `Get` returns a pointer to an object...

> ...thought was that now depending if I list or if I get a policy, I (the developer) have to tread the MonitorScript (or whatever) struct differently

> ...also you can’t sort an array of structs, you can sort an array of pointers

Reply:
👍